### PR TITLE
feat (#157): Adapt gravatar for get user

### LINKED ISSUE #157: [Users] Adapt gravatar avatar for get user / get users, etc methods that uses User entity
For now, any request that returns `User` doesn't know anything about gravatar avatar. So, you need to change it. I recommend to make avatar via file id \ gravatar email as `oneof` as it will look better.

### PR DESCRIPTION
**Pull request by sliderzxc**
close #157

### CODE REVIEW

File: core/src/main/kotlin/io/timemates/backend/users/types/User.kt [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
-import io.timemates.backend.users.types.value.UserName
+import io.timemates.backend.users.types.value.*
 
 data class User(
     val id: UserId,
     val name: UserName,
     val emailAddress: EmailAddress?,
     val description: UserDescription?,
     val avatarId: FileId?,
+    val gravatarId: GravatarId?
```
Comment (y9vad9) [inline core/src/main/kotlin/io/timemates/backend/users/types/User.kt:12]: I think it's better to have sealed there, instead of two dedicated fields

**Review by y9vad9 — APPROVED** [review]

File: infrastructure/grpc-api/src/main/kotlin/io/timemates/backend/services/users/GrpcUsersMapper.kt
Comment (y9vad9) [inline infrastructure/grpc-api/src/main/kotlin/io/timemates/backend/services/users/GrpcUsersMapper.kt:1]: forgot to map here to grpc entity

**Review by y9vad9 — CHANGES_REQUESTED** [review]

File: data/src/main/kotlin/io/timemates/backend/data/users/UserEntitiesMapper.kt [hunk 2]
```diff
@@ -17,21 +14,26 @@ class UserEntitiesMapper {
...
     }
 
     fun toCachedUser(pUser: PostgresqlUsersDataSource.User): CachedUsersDataSource.User = with(pUser) {
-        return CachedUsersDataSource.User(userName, userShortDesc, userAvatarFileId, userEmail)
+        return CachedUsersDataSource.User(userName, userShortDesc, userAvatarFileId, userGravatarId, userEmail)
     }
 
     fun toPostgresqlUserPatch(patch: User.Patch) = with(patch) {
-        PostgresqlUsersDataSource.User.Patch(name?.string, shortBio?.string, avatarId?.string)
+        PostgresqlUsersDataSource.User.Patch(name?.string, shortBio?.string, avatarId?.string, gravatarId?.string)
```
Comment (y9vad9) [inline data/src/main/kotlin/io/timemates/backend/data/users/UserEntitiesMapper.kt:27]: trailing comma

File: data/src/main/kotlin/io/timemates/backend/data/users/datasource/PostgresqlUsersDataSource.kt [hunk 3]
```diff
@@ -78,12 +84,13 @@ class PostgresqlUsersDataSource(private val database: Database) {
...
         val userShortDesc: String?,
         val userAvatarFileId: String?,
-        val gravatarId: String?,
+        val userGravatarId: String?,
     ) {
         data class Patch(
             val userName: String? = null,
             val userShortDesc: String? = null,
             val userAvatarFileId: String? = null,
+            val userGravatarId: String? = null
```
Comment (y9vad9) [inline data/src/main/kotlin/io/timemates/backend/data/users/datasource/PostgresqlUsersDataSource.kt:93]: trailing comma

File: infrastructure/grpc-api/src/main/proto/io/timemates/api/users/types/User.proto [hunk 4]
```diff
@@ -12,23 +12,26 @@ message User {
    * String identifier of 128 length. Can be null
    * if there is no avatar.
    */
-  optional string avatarId = 2;
+  oneof avatar {
+    string avatarId = 2;
+    string gravatarId = 3;
+  }
```
Comment (y9vad9) [inline infrastructure/grpc-api/src/main/proto/io/timemates/api/users/types/User.proto:18]: you're missing optional (I am not sure is there any way to check whether oneof can be checked for optionality, but tryna google it)

**Review by y9vad9 — APPROVED** [review]

**Review by y9slavabogu — APPROVED** [review]

**Pull request merged**